common/hook.h

```diff
@@ -21,6 +21,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include "keyboard.h"
 #include "led.h"
 
+typedef struct _host_driver_t host_driver_t;
+
```

Comment:
Well, I'm not sure if it's my C++ background or not, but in C++ the general rule is to forward declare everything you can, in order to avoid having to include files. This can greatly speed up the compilation, especially for big software. In C++ it would be slightly less verbose though ´struct host_driver_t´ would be enough.

I figured the same rules would apply to C coding too, but maybe not, since due to the less strict typing it might be dangerous, if the warning levels are not high enough. So if you want, I can replace it by an include.